Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -362,18 +363,31 @@ - (BOOL) connectToHostWithName: (NSString*)hostname
     const char* cHost = [hostname cStringUsingEncoding: NSUTF8StringEncoding];
     const char* cPort = [[NSString stringWithFormat: @"%ld", (long)port]
                          cStringUsingEncoding: NSUTF8StringEncoding];
-    if (getaddrinfo(cHost, cPort, &hints, &addr)) {
-        NSString* m = [NSString stringWithFormat: @"failed to get address info, %s", strerror(errno)];
-        return [self populateError: error errNo: errno msg: m];
+    int res = getaddrinfo(cHost, cPort, &hints, &addr);
+    if (res) {
+        NSString* msg = [NSString stringWithFormat: @"failed(%d) to get address info: %s", res,
+                         gai_strerror(res)];
+        if (res == EAI_NONAME) {
+            CBLWarnError(WebSocket, @"%@: %@", self, msg);
+            if (error) {
+                *error = [NSError errorWithDomain: (id)kCFErrorDomainCFNetwork
+                                             code: kCFHostErrorUnknown
+                                         userInfo: @{NSLocalizedDescriptionKey : msg,
+                                                     (id)kCFGetAddrInfoFailureKey: @EAI_NONAME}];
+            }
+        } else
+            [self populateError: error errNo: res msg: msg];
```

Comment:
No need to check `if (res == EAI_NONAME)`. Just convert all EAI Error. CBLStatus will taken care of the error mapping from getaddrinfo().

```
    if (res) {
        NSString* msg = [NSString stringWithFormat: @"failed(%d) to get address info: %s", res, gai_strerror(res)];
        CBLWarnError(WebSocket, @"%@: %@", self, msg);
        if (error) {
                *error = [NSError errorWithDomain: (id)kCFErrorDomainCFNetwork
                                                                  code: kCFHostErrorUnknown
                                                             userInfo: @{NSLocalizedDescriptionKey : msg, (id)kCFGetAddrInfoFailureKey: @(res}];
         }
         return false;
    }
```